prac_09/taxi_simulator.py

```diff
@@ -0,0 +1,54 @@
+"""CP1404 Taxi Simulator Program Prac 09."""
+
+from prac_09.silver_service_taxi import SilverServiceTaxi
+from prac_09.taxi import Taxi
+
+MENU = "q)uit, c)hoose taxi, d)rive"
+
+
+def main():
+    """Runs Taxi Simulator Program."""
+    total_bill = 0
+    taxis = [Taxi("Prius", 100), SilverServiceTaxi("Limo", 100, 2), SilverServiceTaxi("Hummer", 200, 4)]
+    current_taxi = None
+    print("Let's drive!")
+
+    print(MENU)
+    choice = input(">>> ").lower()
+    while choice != "q":
+        if choice == "c":
+            print("Taxis available: ")
+            display_taxis(taxis)
+            taxi_choice = int(input("Choose Taxi: "))
+            try:
+                current_taxi = taxis[taxi_choice]
+            except IndexError:
+                print("Invalid taxi choice")
+        elif choice == "d":
+            if current_taxi:
+                current_taxi.start_fare()
+                distance_to_drive = float(input("How far to drive? "))
+                current_taxi.drive(distance_to_drive)
+                trip_cost = current_taxi.get_fare()
+                print(f"Your {current_taxi.name} trip cost you ${trip_cost:.2f}")
+                total_bill += trip_cost
+            else:
+                print("You need to choose a taxi before you can drive")
+        else:
+            print("Invalid option")
+        print(f"Bill to date: ${total_bill:.2f}")
+        print(MENU)
+        choice = input(">>> ").lower()
+
+    print(f"Total trip cost: ${total_bill:.2f}")
+    print("Taxis are now:")
+    display_taxis(taxis)
+
+
+def display_taxis(taxis):
+    """Display numbered list of taxis."""
+    for i, taxi in enumerate(taxis):
```

Comment:
I wonder if numbering something as '0' is confusing for a non-programmer. Numbering the taxis from 1 to n might be ever so slightly more user friendly than 0 to n. You could do `enumerate(taxis, 1)` and make sure to -1 to the index later.